draft-ietf-mls-protocol.md

```diff
@@ -975,6 +976,26 @@ used to add new members).
 
 The ciphersuites are defined in section {{mls-ciphersuites}}.
 
+## Hash-Based Identifiers
+
+Some MLS messages refer to other MLS objects by hash.  For example, Welcome
+messages refer to KeyPackages for the members being welcomed, and Commits refer
+to Proposals they cover.  These identifiers are computed as follows:
+
+~~~~~
+opaque HashReference[16];
+
+MakeHashRef(value) = KDF.expand(KDF.extract("", value), "MLS 1.0 ref", 16)
+
+HashReference KeyPackageRef;
+HashReference ProposalRef;
+~~~~~
+
+For a KeyPackageRef, the `value` input is the encoded KeyPackage, and the
+ciphersuite specified in the KeyPackage determines the KDF used.  For a
+ProposalRef, the `value` input is the MLSPlaintext carrying the proposal, and
```

Comment:
Should the `Commit` section then use `ProposalRef` rathen than `opaque hash<0..255>` for the `reference`?

Reply:
This is done via the changes to ProposalOrRef further down.

Reply:
Ah missed that, never mind 🤦🏻